examples/react-klevu-ui-v2/src/App.tsx

```diff
@@ -0,0 +1,281 @@
+import {
+  KlevuButton,
+  KlevuDrawer,
+  KlevuIcon,
+  KlevuInit,
+  KlevuPopup,
+  KlevuProduct,
+  KlevuQuicksearch,
+  KlevuSearchField,
+} from "@klevu/ui-react"
+import KlevuLogo from "/klevu.svg"
+import {
+  NavLink,
+  Outlet,
+  useLocation,
+  useNavigate,
+  useSearchParams,
+} from "react-router-dom"
+import { BANNER_ID, CATEGORY_ID, SEARCH_API_KEY, SEARCH_URL } from "./constants"
+import { RefObject, useEffect, useMemo, useRef, useState } from "react"
+import { useCart } from "./cartContext"
+import { createRoot } from "react-dom/client"
+import { KlevuConfig } from "@klevu/core"
+
+function App() {
+  const location = useLocation()
+  const navigate = useNavigate()
+  const [searchParams] = useSearchParams()
+  const [searchSuggestions, setSearchSuggestions] = useState([])
+  const apiKeyRef = useRef<HTMLInputElement>()
+  const drawerRef = useRef<HTMLKlevuDrawerElement>()
+  const popupRef = useRef<HTMLKlevuPopupElement>()
+  const storeUrlRef = useRef<HTMLInputElement>()
+  const recsIdRef = useRef<HTMLInputElement>()
+  const categoryRef = useRef<HTMLInputElement>()
+  const cart = useCart()
+  const updateStoreSettings = () => {
+    navigate(
+      `/?apiKey=${apiKeyRef.current.value}&url=${storeUrlRef.current.value}&categoryId=${categoryRef.current.value}&recsId=${recsIdRef.current.value}`,
+    )
+    window.location.reload()
+  }
+  useEffect(() => {
+    console.log("App useEffect")
+  }, [])
+  const settings = {
+    onItemClick(product, event) {
+      navigate(`/products/${product.itemGroupId}/${product.id}`)
+      event.preventDefault()
+      return false
+    },
+    generateProductUrl(product) {
+      return `/products/${product.itemGroupId}/${product.id}`
+    },
+    renderPrice(amount, currency) {
+      return `${currency} -- ${amount}`
+    },
+  }
+  const apiKey = searchParams.get("apiKey") || SEARCH_API_KEY
+  const url = searchParams.get("url") || SEARCH_URL
+  const recsId = searchParams.get("recsId") || BANNER_ID
+  const categoryId = searchParams.get("categoryId") || CATEGORY_ID
+  useMemo(() => {
```

Comment:
This could be moved out from the component itself. Just before the function call so memo wouldn't need to be used.

Reply:
I had it here since its getting the url and apiKey from searchParams, do you suggest to move out and read from URLSearchParams?

Reply:
Ah I see. This could be used to show how to use React tools. So it's fine.